test/cpp/tensorexpr/test_external_calls.cpp

```diff
@@ -249,6 +251,162 @@ TEST(ExternalCall, Addmm_float) {
   ASSERT_TRUE(at::allclose(nnc_result, ref));
 }
 
+#ifdef USE_XNNPACK
+
+// NOLINTNEXTLINE(cppcoreguidelines-avoid-non-const-global-variables)
+TEST(ExternalCall, Prepacked_Linear_float) {
+  using namespace at::native::xnnpack;
+
+  KernelScope kernel_scope;
+
+  Placeholder Input("Input", kFloat, {100, 200});
+  Placeholder Weight("Weight", kFloat, {300, 200});
+  Placeholder Bias("Bias", kFloat, {300});
+  BufHandle ResultBuf("Result", {100, 300}, kFloat);
+
+  // Calculate reference result using at::linear.
+  auto options = at::TensorOptions()
+                     .dtype(at::kFloat)
+                     .layout(at::kStrided)
+                     .device(at::kCPU)
+                     .requires_grad(false);
+  at::Tensor input = at::ones({100, 200}, options) * 5.f;
+  at::Tensor weight = at::ones({300, 200}, options) * 6.f;
+  at::Tensor bias = at::ones({300}, options) * 11.f;
```

Comment:
these are not needed

also, does the input need to have the right shapes if it's meant as a placeholder?

Reply:
deleted - right, I don't think the input shape has to be real for this case.